File: Structural/Lab10/C#/Entities/Decorators/InteractiveStorytellingDecorator.cs
using System;
using System.Collections.Generic;
using System.Text;
using C_.Interfaces;

namespace C_.Entities.Decorators
{
    public class InteractiveStorytellingDecorator : RealEstateExperienceDecorator
    {
        public InteractiveStorytellingDecorator(IRealEstateExperience realEstate) : base(realEstate) { }

        public override string CustomizeExperience()
        {
            return _realEstate.CustomizeExperience() + "\nOffers an interactive storytelling experience that showcases the history and unique aspects of the property.";
        }
    }
}

File: Structural/Lab10/Typescript/src/Entities/Decorators/interactive_storytelling_decorator.ts
import { IRealEstateExperience } from "../../Interfaces/real_estate_experience";
import { RealEstateExperienceDecorator } from "./real_estate_experience_decorator";

export class InteractiveStorytellingDecorator extends RealEstateExperienceDecorator {

    constructor(realEstate: IRealEstateExperience) {
        super(realEstate);
    }
    public customizeExperience(): String {
        return this._realEstate.customizeExperience() + "\nOffers an interactive storytelling experience that showcases the history and unique aspects of the property.";
    }
}